src/linearoperators/LBFGS.jl

```diff
@@ -30,14 +30,32 @@ mutable struct LBFGS{R, T <: BlockArray, M} <: LinearOperator
 	H::R
 end
 
-function LBFGS(x::T, M::Integer) where {R, T <: BlockArray{R}}
```

Comment:
this created problems as `R` for example when typing `LBFGS( (zeros(3),zeros(Complex{Float64})), 3)'. 
While `typeof( (zeros(3),zeros(Complex{Float64})) ) <: BlockArray{Float64} == true` apparently it couldn't be determined that `R` was `Float64`.